### PR TITLE
Fix #3106: move buf.yml file to /.github/workflows/

### LINKED ISSUE #3106: Move buf.yml file to model package
Move https://github.com/oppia/oppia-android/blob/develop/buf.yaml to https://github.com/oppia/oppia-android/tree/develop/model and ensure nothing gets broken in CI check.

### PR DESCRIPTION
**Pull request by AkinchanKushwaha**
## Explanation

Fixes #3106: Moves buf.yml file to /.github/workflows/'

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by AkinchanKushwaha** [comment]
@anandwana001 How do I fix "Static Checks / Check CODEOWNERS & Repository files (pull_request)" test?

File: scripts/buf_lint_check.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ populate_jar_config_file_paths() {
 
   if [ $github_actions_path ]; then
     jar_file_path="$github_actions_path/oppia-android-tools/$buf_file_name"
-    config_file_path="/home/runner/work/oppia-android/oppia-android/buf.yaml"
+    config_file_path="/home/runner/work/oppia-android/oppia-android/.github/workflows/buf.yaml"
   else
     jar_file_path="../oppia-android-tools/$buf_file_name"
     config_file_path="buf.yaml"
```
Comment (anandwana001) [inline scripts/buf_lint_check.sh]: I think you need to update this too, let's try and see if it works 

```suggestion
    config_file_path=".github/workflows/buf.yaml"
```

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: scripts/buf_lint_check.sh (hunk 1 above)
Comment (AkinchanKushwaha) [reply scripts/buf_lint_check.sh]: I have updated the file path at line 42 as well.

**Comment by AkinchanKushwaha** [comment]
@anandwana001 Any idea on how to fix `Build Tests / Build Binary with Bazel (ubuntu-18.04) (pull_request)` test?

**Review by Sparsh1212 — COMMENTED** [review]
@AkinchanKushwaha Can you please re-write the PR explanation to:
'Fixes #3106: Moves buf.yml file to /.github/workflows/' and also mark the corresponding checkbox of including 'Fixes #BUG_NUMBER' in your PR description.
Everything else LGTM!
Thanks.

**Review by Sparsh1212 — APPROVED** [review]

**Comment by AkinchanKushwaha** [comment]
I have updated the PR description. 
> Can you please re-write the PR explanation

**Review by seanlip — APPROVED** [review]
LGTM for codeowner file.

**Comment by anandwana001** [comment]
@AkinchanKushwaha The lint check is passing successfully but it looks like the GitHub actions is treating the buf file as an action because we are keeping it in the workflows folder. I didn't catch this case before filing the issue. We need to look into where could be a better place where we can keep this configuration file `buf.yaml`. 
Could you recommend any better path to keep this file?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
@AkinchanKushwaha The lint check is passing successfully but it looks like the GitHub actions is treating the buf file as an action because we are keeping it in the workflows folder. I didn't catch this case before filing the issue. We need to look into where could be a better place where we can keep this configuration file `buf.yaml`. 
Could you recommend any better path to keep this file?

**Comment by anandwana001** [comment]
closing this PR as no further update.

**Pull request closed without merging**